source/core/slang-token-reader.h

```diff
@@ -189,6 +189,15 @@ namespace Slang
         {
             tokenPtr -= count;
         }
+        Token ReadMatchingToken(TokenType type)
+        {
+            auto token = ReadToken();
+            if (token.Type != type)
+            {
+                throw TextFormatException("Text parsing error: unexpected token.");
```

Comment:
Hmm. There are already facilities elsewhere in the code for doing this kind of operation (e.g., in the parser) so it seems a bit weird to make this a fundamental service on the token reader when its behavior is not useful in most cases.

* I would expect the function to only output the error to something like a `DiagnosticSink` and not throw an exception (malformed text input is something expected when parsing, not an unexpected problem). At the very least, throwing an exception is a policy decision that would need to be made at a level above the reader, and not baked into its design.

* I'd expect this function to *only* consume a token from the input if it matched the expected `type`, and otherwise to leave the input alone. This obviously matters less when we are `throw`ing, but it matters a lot more when doing any more advanced error reporting and recovery

* As a small note (and I don't think the Slang codebase follows this right now, due to legacy) I tend to like a function like this to have the verb be `expect` rather than `read`, so you have code like:

```
if(/* detect that you have some construct*/)
{
    tokenStream.expect(TokenType::LParen);
    ...= readTheStuffBetweenParens(tokenStream);
    tokenStream.expect(TokenType::RParen);
}
```

That kind of logic tends to read very cleanly at the site that is implementing parsing logic.